src/watertap_contrib/reflo/code_demos/steady_state_flowsheets/battery.py

```diff
@@ -0,0 +1,284 @@
+#################################################################################
+# DISPATCHES was produced under the DOE Design Integration and Synthesis Platform
+# to Advance Tightly Coupled Hybrid Energy Systems program (DISPATCHES), and is
+# copyright (c) 2020-2023 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#################################################################################
+import sys
+from pandas import DataFrame
+from collections import OrderedDict
+import textwrap
+
+# Import Pyomo libraries
+from pyomo.environ import Var, Param, NonNegativeReals, units as pyunits, value
+from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import declare_process_block_class, UnitModelBlockData
+from idaes.core.solvers import get_solver
+from idaes.core.util.tables import stream_table_dataframe_to_string
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_activated_constraints,
+    number_activated_blocks,
+)
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("BatteryStorage", doc="Simple battery model")
+class BatteryStorageData(UnitModelBlockData):
```

Comment:
Since this is a unit model, it really belongs in our `unit_models` folder rather than `code_demos/steady_state_flowheets`.

Reply:
I moved it into the zero_order unit model folder